### ISSUE TITLE
[rllib] [feature request] Add or expand method of agents to compute deterministic actions

### ISSUE THREAD

**Issue by FedericoFontana** [ray-project/ray#5729]
# System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Ray installed from (source or binary)**: pip install ray
- **Ray version**: 0.7.4
- **Python version**: 3.6

### Describe the problem
After having successfully trained and restore an agent, one very common use case might be to  make deterministic action given a state. After training, or when deploying an agent in production, one should make deterministic actions by default, unless there is a valid reason to add a bit of stochasticity (e.g. in Minh 2015, 2% of actions are random to escape from hero getting stuck in corner cases I seem to rememeber). Deterministic actions have the benefit of optimizing the present value of future rewards in production, whereas stochastic actions explore the environment, which is not necessarily needed.

As of now, one could (kind of) access deterministic actions retrieving the logits (i.e. deterministic) of the network with `agent.get_policy().compute_single_action(state)['behaviour_logits']`. However, how to map behaviour_logits to action is non-trivial as ray make different assumptions on what the logits should represent, depending on the action_space, whether or not `config['use_lstm']=True` and possibly other conditions. All this is too low-level and not a pleasant UX for such common use case I believe.

The requirement described in this issue can be addressed implementing a new method or expanding existing methods as indicated in the section `# Requirement` below.

### Source code
```
import gym
import ray
from ray import rllib, tune
info = ray.init()

# Instance an agent.
env = gym.make('MountainCarContinuous-v0')
config = rllib.agents.ppo.DEFAULT_CONFIG.copy()
#config['model']['use_lstm'] = True
agent = rllib.agents.ppo.PPOTrainer(
    env='MountainCarContinuous-v0',
    config=config,
)

# Compute an action
state = env.reset()
non_deterministic_action = agent.compute_action(state)

# Requirement. This should ideally work also when using RNN.
# deterministic_action = agent.compute_action(state, deterministic=True)
#     OR
# deterministic_action = agent.compute_deterministic_action(state)


```

**Comment by ericl** [comment]
I agree this would be a nice feature. Perhaps it could be part of the agent config, e.g. exploration: False?

Contributions would be welcome here.

**Comment by FedericoFontana** [comment]
I'm afraid that I don't have time to contribute for now I'm afraid. The solution with Simplex is to take softmax of logits, whereas with Box the deterministic action corresponds to the first half of logits as the second half represent the variance of the diagonal multivariate Gaussian distribution. I don't know if this is true for agents different from PPO.

**Comment by anooprh** [comment]
@ericl 
I see this is labelled with `good first issue`. I'm interested in contributing to this. 

I'm a novice w.r.t RL as well as ray framework, but want to use the opportunity to get familiar with both. I will read up some more about stochastic actions v/s deterministic actions and try to post my understanding of the problem + the fix for this.

**Comment by anooprh** [comment]
I'm able to recreate the scenario which @FedericoFontana is facing. 

But I'm having a tough time understanding the types of `non_deterministic_action` , and also the logic used to decide the `detrministic_action`.

For e.g in the code snippet below,

```
import gym
import ray
from ray import rllib, tune
info = ray.init()

# Instance an agent.
env = gym.make('MountainCarContinuous-v0')
config = rllib.agents.ppo.DEFAULT_CONFIG.copy()
#config['model']['use_lstm'] = True
agent = rllib.agents.ppo.PPOTrainer(
    env='MountainCarContinuous-v0',
    config=config,
)
state = env.reset()
non_deterministic_action = agent.compute_action(state)
# array([-1.], dtype=float32)

agent.get_policy().compute_single_action(state=[], obs=state)
# (array([0.14374927], dtype=float32),
# [],
# {'action_prob': 0.39686415,
#  'action_logp': -0.92416126,
#  'vf_preds': 0.0009352048,
# 'behaviour_logits': array([ 0.00126918, -0.00503013], dtype=float32)})
```
I was expecting `non_deterministic_action` to be an array of size representing all possible action items (2 in this case as the action is to go either left or right), but it's an array of size 1. How do I interpret this ? 

Secondly, for computing the deterministic action, I assume we can take a softmax of the `behaviour_logits` to generate the deterministic action, but is the `compute_single_action` always guaranteed to have the logits or are there other ways in which an action is computed ?

cc : @ericl
